pkg/scd/store/cockroach/store.go

```diff
@@ -13,33 +15,85 @@ var (
 	DefaultClock = clockwork.NewRealClock()
 )
 
-// Store is an implementation of dss.Store using
-// Cockroach DB as its backend store.
+// Store is an implementation of scd.Store using
+// a CockroachDB transaction.
 type Store struct {
-	*cockroach.DB
+	tx     *sql.Tx
+	ctx    context.Context
```

Comment:
Please don't store contexts within structs, as it makes overriding contexts in subcalls much more difficult to reason about.
https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39
https://www.ardanlabs.com/blog/2019/09/context-package-semantics-in-go.html
http://p.agnihotry.com/post/understanding_the_context_package_in_golang/

Reply:
It seems unlikely we would want to override a context for a single query within a transaction, but will do since an explicit context parameter seems idiomatic to Go, [http/request](https://golang.org/src/net/http/request.go) notwithstanding.

Reply:
Done (4b234b8).